From 334359846401d82865b74291d012bda270d295d3 Mon Sep 17 00:00:00 2001
From: Ganesh Kariganuru Mahabalesh <gkarigan@qti.qualcomm.com>
Date: Mon, 24 Nov 2025 15:53:36 -0800
Subject: [PATCH] QSDK:CP:iw:Add vendor data attribute for interface offload
 type in GET handler

Currently, the GET interface offload handler did not include any vendor data
attributes, which could lead to incomplete or invalid requests when querying
offload configuration.

Add a nested NL80211_ATTR_VENDOR_DATA attribute and set
QCA_WLAN_VENDOR_ATTR_IF_OFFLOAD_TYPE to 0 to ensure the request includes the
required vendor-specific information.

Patch-dependency:
	622-QSDK-iw-wifi-Add-support-for-get-the-interface-offload.patch 

Signed-off-by: Ganesh Kariganuru Mahabalesh <gkarigan@qti.qualcomm.com>
---
 set_interface_offload.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/set_interface_offload.c
+++ b/set_interface_offload.c
@@ -141,10 +141,17 @@ static int handle_get_intf_offload(struc
 				   int argc, char **argv,
 				   enum id_input id)
 {
+	struct nlattr *offload_type;
+
 	nla_put_u32(msg, NL80211_ATTR_VENDOR_ID, OUI_QCA);
 	nla_put_u32(msg, NL80211_ATTR_VENDOR_SUBCMD,
 		    QCA_NL80211_VENDOR_SUBCMD_GET_WIFI_CONFIGURATION);
 
+	offload_type = nla_nest_start(msg, NL80211_ATTR_VENDOR_DATA);
+	nla_put_u8(msg, QCA_WLAN_VENDOR_ATTR_IF_OFFLOAD_TYPE, 0);
+
+	nla_nest_end(msg, offload_type);
+
 	register_handler(print_intf_offload_type, NULL);
 	return NL_OK;
 }
